protos/sigstore_trustroot.proto

```diff
@@ -43,22 +45,34 @@ message TransparencyLogInstance {
         // calculated over the DER encoding of the key represented as
         // SubjectPublicKeyInfo.
         // See https://www.rfc-editor.org/rfc/rfc6962#section-3.2
-        dev.sigstore.common.v1.LogId log_id = 4;
-        // The checkpoint key identifier for the log used in a checkpoint.
-        // Optional, not provided for logs that do not generate checkpoints.
-        // For logs that do generate checkpoints, if not set, assume
-        // log_id equals checkpoint_key_id.
-        // Follows the specification described here
-        // for ECDSA and Ed25519 signatures:
+        // MUST set checkpoint_key_id if multiple logs use the same
+        // signing key.
+        // Deprecated: Use checkpoint_key_id instead, since log_id is not
+        // guaranteed to be unique across multiple deployments. Clients
+        // must use the key name and key ID from a checkpoint to determine
+        // the correct TransparencyLogInstance to verify a proof.
+        dev.sigstore.common.v1.LogId log_id = 4 [deprecated = true];
+        // The unique identifier for the log, used in the checkpoint.
+        // Its calculation is described in
         // https://github.com/C2SP/C2SP/blob/main/signed-note.md#signatures
-        // For RSA signatures, the key ID will match the ECDSA format, the
-        // hashed DER-encoded SPKI public key. Publicly witnessed logs MUST NOT
-        // use RSA-signed checkpoints, since witnesses do not support
-        // RSA signatures.
+        // SHOULD be set for all logs. When not set, clients MUST use log_id.
+        //
+        // For Ed25519 signatures, the key ID is computed per the C2SP spec:
+        // key ID = SHA-256(key name || 0x0A || 0x01 || 32-byte Ed25519 public key)[:4]
+        // For ECDSA signatures, the key ID is computed per the C2SP spec:
+        // key ID = SHA-256(PKIX ASN.1 DER-encoded public key, in SubjectPublicKeyInfo format)[:4]
+        // For RSA signatures, the signature type will be 0xff with an appended identifier for the format,
+        // "PKIX-RSA-PKCS#1v1.5":
+        // key ID = SHA-256(key name || 0x0A || 0xff || PKIX-RSA-PKCS#1v1.5 || PKIX ASN.1 DER-encoded public key)[:4]
+        //
         // This is provided for convenience. Clients can also calculate the
         // checkpoint key ID given the log's public key.
-        // SHOULD be set for logs generating Ed25519 signatures.
         // SHOULD be 4 bytes long, as a truncated hash.
+        //
+        // To find a matching TransparencyLogInstance in the TrustedRoot,
+        // clients will parse the checkpoint, and for each signature line,
+        // extract the key name (i.e. log origin, base_url) and 4-byte key ID,
+        // and compare the key name and key ID against each log instance.
```

Comment:
maybe avoid saying that key name is compared since it seems only the keyid is compared:
```suggestion
        // use the key name (i.e. log origin, base_url) and public key to calculate
        // key ID which can then be compared against the TrustedRoot log instances
```

Reply:
Thanks, incorporated!

Reply:
Actually sorry, I need to revert this partially. The client wouldn't know the log public key yet. The client only has what the checkpoint provides, which is the origin and key ID in the checkpoint signature line. The client can verify that the key ID is properly computed if it wants to after it retrieves the public key from the TrustedRoot, but I don't think that's strictly necessary.